src/containers/OasisTrade.test.js

```diff
@@ -1,22 +1,275 @@
 /* global shallow describe it expect */
-/* eslint-disable import/first */
+/* eslint-disable import/first,no-undef */
 import React from 'react';
 import Immutable from 'immutable';
+import { shallow } from 'enzyme';
+
 
 import {
-  OasisTradeWrapper,
-  mapStateToProps,
-  mapDispatchToProps,
+OasisTradeWrapper,
+mapStateToProps,
+mapDispatchToProps,
 } from './OasisTrade';
-import { shallow } from 'enzyme';
 
 describe('(Container) OasisTrade', () => {
-  const state = Immutable.fromJS({});
+  const state = Immutable.fromJS({
+    tokens: {
+      allTokens: [
+        'W-ETH',
+        'MKR',
+        'DGD',
+        'GNT',
+        'W-GNT',
+        'REP',
+        'ICN',
+        '1ST',
+        'SNGLS',
+        'VSL',
+        'PLU',
+        'MLN',
+        'RHOC',
+        'TIME',
+        'GUP',
+        'BAT',
+        'NMR',
+        'SAI',
+      ],
+      baseTokens: [
+        'W-GNT',
+        'DGD',
+        'REP',
+        'ICN',
+        '1ST',
+        'SNGLS',
+        'VSL',
+        'PLU',
+        'MLN',
+        'RHOC',
+        'TIME',
+        'GUP',
+        'BAT',
+        'NMR',
+      ],
+      quoteTokens: [
+        'W-ETH',
+      ],
+      tradingPairs: [
+        {
+          base: 'MKR',
+          quote: 'W-ETH',
+          priority: 10,
+        },
+        {
+          base: 'W-ETH',
+          quote: 'SAI',
+          priority: 9,
+        },
+        {
+          base: 'MKR',
+          quote: 'SAI',
+          priority: 8,
+        },
+        {
+          base: 'W-GNT',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'DGD',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'REP',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'ICN',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: '1ST',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'SNGLS',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'VSL',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'PLU',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'MLN',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'RHOC',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'TIME',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'GUP',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'BAT',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+        {
+          base: 'NMR',
+          quote: 'W-ETH',
+          priority: 0,
+        },
+      ],
+      tokenSpecs: {
```

Comment:
For testing purposes you don't need the actual data right.. You can provide 2 - 3 pairs with different priority and test that.  
Actually you will have one test scenario which you test the  sorting by priority so in your setup you can have 2 pairs with 2 different priorities and that's it and another test scenario where  you have two pairs with same priority different volume

Reply:
We only test bare functionality required  for the component to bootstrap at this moment.